src/wallet/rpcwallet.cpp

```diff
@@ -135,14 +135,15 @@ UniValue getnewaddress(const JSONRPCRequest& request)
         return NullUniValue;
     }
 
-    if (request.fHelp || request.params.size() > 1)
+    if (request.fHelp || request.params.size() > 2)
         throw std::runtime_error(
-            "getnewaddress ( \"account\" )\n"
+            "getnewaddress ( \"account\" \"address_type\" )\n"
             "\nReturns a new Bitcoin address for receiving payments.\n"
             "If 'account' is specified (DEPRECATED), it is added to the address book \n"
             "so payments received with the address will be credited to 'account'.\n"
             "\nArguments:\n"
             "1. \"account\"        (string, optional) DEPRECATED. The account name for the address to be linked to. If not provided, the default account \"\" is used. It can also be set to the empty string \"\" to represent the default account. The account does not need to exist, it will be created if there is no account by the given name.\n"
+            "2. \"address_type\"   (string, optional) The address type to use. Options are \"legacy\", \"p2sh\", and \"bech32\". Default is set by -addresstype.\n"
```

Comment:
There's no indication as to what "-addresstype" currently is. How about something like:
```c++
"... Default (*g_address_type string*) is set by -addresstype.\n"
```

Reply:
Fixed.

Reply:
I fixed it and changed it back in response to a comment by @TheBlueMatt.

Reply:
Should use same format as others?
```cpp
            "2. \"address_type\"   (string, optional, default=" + FormatOutputType(g_address_type) + ") The address type to use...
```
Nit, drop command line argument hint, IMO.